File: lib/models/todo.dart
class Todo {
  String? id;
  String title;
  bool isDone;
  DateTime? createdAt;
  DateTime? updatedAt;
  int position; // Add this field

  Todo({
    this.id,
    required this.title,
    required this.isDone,
    this.createdAt,
    this.updatedAt,
    this.position = 0, // Default value
  });

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['id'],
      title: json['title'],
      isDone: json['done'] ?? false,
      createdAt:
          json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt:
          json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
      position: json['position'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'done': isDone,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
      'position': position,
    };
  }
}

File: Backend/todo/src/main/java/com/spring_boot_mongodb/todo/model/Todo.java
package com.spring_boot_mongodb.todo.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document(collection = "todos")
public class Todo {
    @Id
    private String id;
    private String title;
    private boolean isDone;
    private Date createdAt;
    private Date updatedAt;
    private int position; // Add this field

    // Getters and Setters
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public boolean isDone() {
        return isDone;
    }

    public void setDone(boolean done) {
        isDone = done;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    // Add getter and setter
    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }
}